FreeRTOS_TCP_WIN.c

```diff
@@ -23,6 +23,8 @@
  * http://www.FreeRTOS.org
  */
 
+#if ( ipconfigUSE_TCP == 1 )
```

Comment:
~~~c
 *
 * http://aws.amazon.com/freertos
 * http://www.FreeRTOS.org
 */

#if ( ipconfigUSE_TCP == 1 )
~~~
Oops, this is much too early in the file. Nothing has been included yet and `ipconfigUSE_TCP` is not yet defined.

I would put the test right after these includes:
~~~c
 /* FreeRTOS+TCP includes. */
 #include "FreeRTOS_UDP_IP.h"
 #include "FreeRTOS_IP.h"
 #include "FreeRTOS_Sockets.h"
 #include "FreeRTOS_IP_Private.h"

-#include "FreeRTOSIPConfigDefaults.h"
+#if ( ipconfigUSE_TCP == 1 )

~~~

and please remove `#include "FreeRTOSIPConfigDefaults.h"`, because that file has already been included in the headers files. it is confusing to repeat it here.

You may try this by replacing `#include` with the following:
~~~c
#ifndef FREERTOS_DEFAULT_IP_CONFIG_H
    #error Uh?
#endif
~~~

Reply:
Ah! I missed that completely.
Done, thank you @htibosch.

Reply:
I had to also uncrustify the files which now show a lot of changes due to that.